pkg/util/iptables/iptables.go

```diff
@@ -783,3 +789,90 @@ func isResourceError(err error) bool {
 	}
 	return false
 }
+
+// ParseError records the payload when iptables restore fails
+type ParseError interface {
+	// Line returns the number of lines on which iptables-restore is failing
+	// NOTE: First line is line 1
+	Line() int
+	// Error returns the error message of the location where the iptables-restore execution failed
+	Error() string
+}
+
+type parseError struct {
+	cmd  string
+	line int
+}
+
+func (e parseError) Line() int {
+	return e.line
+}
+
+func (e parseError) Error() string {
+	return fmt.Sprintf("%s: input error on line %d: ", e.cmd, e.line)
+}
+
+// LineData is used to save the line of the lines and its corresponding data
+type LineData struct {
+	// Number of rows in lines, First line is line 1
+	Line int
+	// The date of the line
```

Comment:
"data" and end with a period, please.

Reply:
my bad😅

Reply:
no problem - we're down to really minor details now :)